jdbc-v2/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java

```diff
@@ -358,30 +369,49 @@ public boolean isValid(int timeout) throws SQLException {
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+            this.defaultQuerySettings.setOption(name, value);
+        } catch (Exception e) {
+            throw new SQLClientInfoException("Failed to set client info.", Collections.singletonMap(name, ClientInfoStatus.REASON_UNKNOWN), e);
+        }
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
-        throw new SQLClientInfoException("ClientInfo not supported", null);
+        try {
+            checkOpen();
+        } catch (SQLException e) {
+            throw new SQLClientInfoException("Failed to set client info.", new HashMap<>(), e);
+        }
+
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            setClientInfo(entry.getKey().toString(), entry.getValue().toString());
+        }
     }
 
     @Override
     public String getClientInfo(String name) throws SQLException {
         checkOpen();
-        return null;
+        Object value = this.defaultQuerySettings.getAllSettings().get(name);
```

Comment:
according to the comment above: we need to return only client info. In this case it will be value of 'User-Agent' header. 
I will do according change in the client.